mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -436,19 +438,15 @@ fn find_pointers_and_order_post_crash(file: &mut File, max_file_size: u64) -> Ri
             read = end + data_size;
         }
 
-        // Found the last write, take whatever we got for read and write
-        // and return the pointers.
+        // Found the last write, can stop updating write.
         if inner.order() < order {
-            return RingBufferMetadata {
-                file_pointers: FilePointers {
-                    write: start,
-                    read_begin: read,
-                    read_end: read,
-                },
-                order,
-                // Check to see if wrap around case.
-                can_read_from_wrap_around: (start == read && write_updates > read_updates),
-            };
```

Comment:
So this was the problem? We stopped this loop at the last write meaning it is possible that the read pointer is incorrect?

Reply:
Yeah, I think this was an issue, not sure if this is relevant to the issue you saw though